drivers/gpio/Kconfig.lpc11u6x

```diff
@@ -0,0 +1,31 @@
+# LPC11U6x GPIO configuration options
+
+# Copyright (c) 2020 Seagate Technology LLC
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig GPIO_LPC11U6X
+	bool "LPC11U6x GPIO driver"
+	depends on SOC_SERIES_LPC11U6X
+	depends on CLOCK_CONTROL_LPC11U6X
+	depends on PINMUX_LPC11U6X
+	help
+	  Enable GPIO driver for LPC11U6x MCUs.
+
+if GPIO_LPC11U6X
+
+config GPIO_LPC11U6X_PORT0
+	bool "Port 0"
+	help
+	  Enable port 0.
+
+config GPIO_LPC11U6X_PORT1
+	bool "Port 1"
+	help
+	  Enable port 1.
+
+config GPIO_LPC11U6X_PORT2
+	bool "Port 2"
+	help
+	  Enable port 2.
```

Comment:
Yes it can. In my understanding (coming from the Linux world), DT is about describing the hardware, and not about configuring it. In Linux there is a strong line about that. So if a board does provide two GPIO ports, then they have to be defined and enabled both in the DT. And the application can choose to use them, or not... With @mbittan we realized this was maybe not the way it was working in Zephyr. But it was pretty late in the process and we chose to keep the GPIO driver with the CONFIG_ port options and an abnormal macros usages to ensure a correct match between GPIO instances and ports numbers. 

So you confirm that the correct way to disable a GPIO port is by disabling its DT node from an application overlay ?

Reply:
Yes, that's how this is being done these days.  It's closer to the Linux way than Kconfig.  Some day we should have the ability to enable/disable at runtime, too, for power saving.

Reply:
We removed the CONFIG_ options and are now using the status to instantiate a GPIO device.